src/util/std_expr.h

```diff
@@ -3094,30 +3086,26 @@ inline void validate_expr(const not_exprt &value)
 
 /*! \brief Operator to dereference a pointer
 */
-class dereference_exprt:public exprt
+class dereference_exprt:public unary_exprt
 {
 public:
-  explicit dereference_exprt(const typet &type):
-    exprt(ID_dereference, type)
+  dereference_exprt():unary_exprt(ID_dereference)
   {
-    operands().resize(1);
   }
 
-  explicit dereference_exprt(const exprt &op):
-    exprt(ID_dereference)
+  explicit dereference_exprt(const typet &type):
+    unary_exprt(ID_dereference, type)
   {
-    copy_to_operands(op);
   }
 
-  dereference_exprt(const exprt &op, const typet &type):
-    exprt(ID_dereference, type)
+  explicit dereference_exprt(const exprt &op):
+    unary_exprt(ID_dereference, op)
   {
-    copy_to_operands(op);
   }
 
-  dereference_exprt():exprt(ID_dereference)
+  dereference_exprt(const exprt &op, const typet &type):
+    unary_exprt(ID_dereference, op, type)
```

Comment:
There should be a check that indeed `type==op.subtype()`

Reply:
Unfortunately still needs the namespace, until we get rid of the symbol_typet

Reply:
Unless I'm mistaken, a pointer would never be a `symbol_typet` (even though the `subtype()` might be one).